src/Shared/MSBuildLoadContext.cs

```diff
@@ -19,9 +19,8 @@ namespace Microsoft.Build.Shared
     internal class MSBuildLoadContext : AssemblyLoadContext
     {
         private readonly string _directory;
-        private readonly object _guard = new object();
 
-        private static readonly ImmutableHashSet<string> _wellKnownAssemblyNames =
+        internal static readonly ImmutableHashSet<string> WellKnownAssemblyNames =
```

Comment:
It generally doesn't need to be because MSBuild is what does the loading and thus doesn't need to be unified. I added it here to smooth out some of our unit tests that do things differently.